reaper/reaper.go

```diff
@@ -371,8 +488,51 @@ func allReapables() (map[string][]reaperevents.Reapable, []reaperevents.Reapable
 			}
 		}
 	}
-	if config.Snapshots.Enabled {
 
+	// get all instances
+	for i := range getInstances() {
+		// add security groups to map of in use
+		for id, name := range i.SecurityGroups {
+			dependency[i.Region][reapable.ID(name)] = true
+			dependency[i.Region][id] = true
+		}
+
+		if isInCloudformation[i.Region][i.ID] {
+			i.IsInCloudformation = true
+		}
+		if instancesInASGs[i.Region][i.ID] {
+			i.AutoScaled = true
+		}
+		// group instances by owner
+		if config.Instances.Enabled {
+			if i.Owner() != nil {
+				owned[i.Owner().Address] = append(owned[i.Owner().Address], i)
+			} else {
+				// if unowned, append to unowned
+				unowned = append(unowned, i)
+			}
+		}
+	}
+
+	// get all security groups
+	for s := range getSecurityGroups() {
+		// if the security group is in use, it isn't reapable
+		// names and IDs are used interchangeably by different parts of the API
+		if isInCloudformation[s.Region][s.ID] {
+			s.IsInCloudformation = true
+		}
+		if dependency[s.Region][s.ID] || dependency[s.Region][reapable.ID(*s.GroupName)] {
```

Comment:
This could be rethought... instead of skipping the SG we could just flag it as being depended on